main.tf

```diff
@@ -81,7 +81,7 @@ module "consul_cluster" {
   cluster_size  = var.consul_cluster_size
   instance_type = var.consul_instance_type
 
-  ami_id = var.ami_id != "" ? var.ami_id : data.aws_ami.vault_consul.image_id
+  ami_id = var.ami_id == null ? data.aws_ami.vault_consul.id : var.ami_id
```

Comment:
if you remove data source, what is the reason of keeping `if ... else ..` condition? Just remove it.